lsp/nls/src/cache.rs

```diff
@@ -72,11 +84,26 @@ impl CacheExt for Cache {
             lin_cache.insert(file_id, linearized);
             Ok(CacheOp::Done(()))
         } else {
-            panic!()
+            unreachable!()
```

Comment:
I suspect this wasn't related to your PR, but while we're at it, could we add an explicit justification of why this is unreachable? I've had to debug unreachable recently and the fact that there wasn't any justification although it was'not trivial was a hinder. I imagine it's because if we could get the term from `self.terms`, then it must have been parsed before, and thus the invariant `*state > EntryState::Typechecked` must hold.

Reply:
Sure 👍. I think it is because `EntryState::Parsed` is the first item of the enum, and we have the case `if *state >= EntryState::Parsed ` before the `else` branch which contains the `uncreachable!()`, so it it actually unreachable unless the order of the enum is changed 😅